scipy/spatial/tests/test_lloyd_algorithm.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+
+from scipy.spatial.distance import cdist
+from scipy.spatial import lloyd_centroidal_voronoi_tessellation
+
+
+def test_lloyd():
+    def l1_norm(points):
+        l1 = cdist(points, points, 'cityblock')
+        return np.min(l1[l1.nonzero()])
```

Comment:
hmm, this same function is defined in the main code as well, isn't there a way to leverage the private function instead of typing it out again here? The privacy of the function is only informal in its name right?

Reply:
Indeed, it's not optimal. I will use the version from the main code. For reference, I have a general wish for such metrics gh-14037

Reply:
Looks to me like this is resolved by `from scipy.spatial._lloyd_algorithm import _l1_norm`.